src/segments/mercurial.go

```diff
@@ -0,0 +1,166 @@
+package segments
+
+import (
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+)
+
+const (
+	MERCURIALCOMMAND = "hg"
+
+	hgLogTemplate = "{rev}|{node}|{branch}|{tags}|{bookmarks}"
+)
+
+type MercurialStatus struct {
+	ScmStatus
+}
+
+func (s *MercurialStatus) add(code string) {
+	switch code {
+	case "R", "!":
+		s.Deleted++
+	case "A":
+		s.Added++
+	case "?":
+		s.Untracked++
+	case "M":
+		s.Modified++
+	}
+}
+
+type Mercurial struct {
+	scm
+
+	Working           *MercurialStatus
+	IsTip             bool
+	LocalCommitNumber string
+	ChangeSetID       string
+	ChangeSetIDShort  string
+	Branch            string
+	Bookmarks         []string
+	Tags              []string
+}
+
+func (hg *Mercurial) Template() string {
+	return "hg {{.Branch}} ({{.LocalCommitNumber}}:{{.ChangeSetIDShort}}){{range .Bookmarks }} \uf02e {{.}}{{end}}{{range .Tags}} \uf02b {{.}}{{end}}{{if .Working.Changed}} \uf044 {{ .Working.String }}{{ end }}" //nolint: lll
+}
+
+func (hg *Mercurial) Enabled() bool {
+	if !hg.shouldDisplay() {
+		return false
+	}
+
+	displayStatus := hg.props.GetBool(FetchStatus, false)
+	if displayStatus {
+		hg.setMercurialStatus()
+	} else {
+		hg.Working = &MercurialStatus{}
+	}
+
+	return true
+}
+
+func (hg *Mercurial) shouldDisplay() bool {
+	if !hg.hasCommand(MERCURIALCOMMAND) {
+		return false
+	}
+
+	hgdir, err := hg.env.HasParentFilePath(".hg")
+	if err != nil {
+		return false
+	}
+
+	if hg.shouldIgnoreRootRepository(hgdir.ParentFolder) {
+		return false
+	}
+
+	hg.setDir(hgdir.ParentFolder)
+
+	hg.workingDir = hgdir.Path
+	hg.rootDir = hgdir.Path
+	// convert the worktree file path to a windows one when in a WSL shared folder
+	hg.realDir = strings.TrimSuffix(hg.convertToWindowsPath(hgdir.Path), "/.hg")
+	return true
+}
+
+func (hg *Mercurial) setDir(dir string) {
+	dir = platform.ReplaceHomeDirPrefixWithTilde(hg.env, dir) // align with template PWD
+	if hg.env.GOOS() == platform.WINDOWS {
+		hg.Dir = strings.TrimSuffix(dir, `\.hg`)
+		return
+	}
+	hg.Dir = strings.TrimSuffix(dir, "/.hg")
+}
+
+func (hg *Mercurial) setMercurialStatus() {
+	hg.Working = &MercurialStatus{}
+	hg.Branch = hg.command
+
+	idString := hg.getHgCommandOutput("log", "-r", ".", "--template", hgLogTemplate)
+	if idString == "" {
```

Comment:
```suggestion
	if len(idString) == 0 {
```

for consistency with everything else